### PR TITLE
Added IP validation

### PR DESCRIPTION
**Pull request by ParthKolekar**
Added a template IP validation on get_magic_user. 
We would be able to return more sensitive data to internal nodes.

### CODE REVIEW

File: src/app/helpers/ip_validation.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
+function check_ipv4_in_cidr($ip, $cidr) {
```
Comment (hharchani) [inline src/app/helpers/ip_validation.php:2]: Dude! You need to open <?php tag!

Comment (ParthKolekar) [inline src/app/helpers/ip_validation.php:2]: Aisa? :stuck_out_tongue:

File: src/app/helpers/ip_validation.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+function check_ipv4_in_cidr($ip, $cidr) {
+    if (false !== strpost($cidr, '/')) {
```
Comment (polybuildr) [inline src/app/helpers/ip_validation.php]: Do you mean `strpos`?

**Comment by ParthKolekar** [comment]
Needless to say, this is not tested.

File: src/app/controllers/auth.php [hunk 3]
```diff
@@ -576,9 +576,14 @@ function get_magic_user() {
...
         $user = $this->auth_model->get_user($oauth_id);
 
+        global $auth_cfg;
+        $remote_ip = $_SERVER['REMOTE_ADDR'];
+        load_helper('ip_validation');
+
         if (!empty($user)
             && isset($user["resitration_status"]) && $user["resitration_status"] == "complete"
             && isset($user["email_verified"]) && $user["email_verified"]
+            && check_ipv4_in_cidr($remote_ip, $auth_cfg['magic_hosts'])
```
Comment (hharchani) [inline src/app/controllers/auth.php:586]: Why do you have magic hosts in config? We can use IP range directly here, no?

Comment (nisargjhaveri) [inline src/app/controllers/auth.php:586]: We try not to use magic constants in code :P

Comment (hharchani) [inline src/app/controllers/auth.php:586]: Aisa :stuck_out_tongue:

**Comment by ParthKolekar** [comment]
Too many implementation problems. Someone who knows php should do this.

**Pull request closed without merging**